### PR TITLE
Redesigning the window geometry/dimension settings

### LINKED ISSUE #336: Redesigning the window geometry/dimension settings
As mentioned in a [comment](https://github.com/dunst-project/dunst/pull/335#issuecomment-315520790) in #335, a lot of the current display settings are built on top of each-other in a confusing way e.g. the height part of the geometry is actually number of notifications to be displayed, `notification_height` is the pixel height, or the confusion between width, dynamic width and `shrink` which takes some close reading of the documentation to understand the difference.

We should
- [ ] Compile a list of use cases we should account for (feel free to post yours if it's not that well supported, or not supported at all)
- [ ] Create a list of changes so that the end result is an intuitive configuration experience while not sacrificing any of the currently available functionality

---

As it stands, geometry is probably the most complicated part of the config since there are a lot of corner cases one might not predict. For example, if the width is omitted the message expands to cover the whole screen but if it's 0 it's considered dynamic and expands to the longest message.

After looking at the current implementation I came up with the following proposal:

Replace `geometry` `shrink`, `notification_height` with the following
* `minimum_width`: The minimum pixel width for the notification as described in #335 
* `minimum_height`: The equivalent of `notification_height`
If any of the minimum_* settings is zero, it is considered dynamic and will be shortened as needed.
* `maximum_width`: The maximum width in pixels for the notification, when that limit is reached it will either be wrapped, or ellipsized according to `word_wrap`
* `maximum_height`: The maximum height in pixels for the notification, after that limit is reached any further text is cut and ellipsized.

* `origin`: A setting that can be set to `topleft`, `bottomleft`, `middle`, `topright` and `bottomright` corresponding to the location that the notification will be drawn at. `middle` would also account for the notification window length to account for proper centering as described in #233 
* `offset_horizontal`: The offset in pixels from the origin in the horizontal axis, can be positive (to the left) or negative (to the right
* `offset_vertical`: As with `offset_horizontal` but in the vertical axis.
* `notification_limit`: The maximum number of notification that will be displayed at any one time (equivalent to geometry's height setting)

Another important point to consider to account for backwards compatibility with the previous settings if possible to provide the best experience for downstream users.

This is a very rough draft at this point and unclear whether any of this is going to move forward. I would love to hear some opinions on this, and more suggestions if you think there is a better way or I missed something.

### LINKED ISSUE #518: Not correctly wrapping text with shrink enabled.
Setting a max width on notifications, along with shrink and word_wrap seems to wrap content based on a notification's min width instead of the width of any already present notifications.

Center aligned
scrot

Right aligned
scrot

### Installation info
- Version: 1.3.2
- Install type: package
- Distro and version: Void Linux

### PR DESCRIPTION
**Pull request by fwSmit**
This PR implements #336, building on top of settings refactor v2
See https://github.com/dunst-project/dunst/issues/336#issuecomment-812845742 for a description of the latest idea
Fixes #336 #82 #233
Seems to fix #518

### CODE REVIEW

**Comment by fwSmit** [comment]
Weird. No CI is running in this PR anymore

**Comment by fwSmit** [comment]
It somehow works again. Sorry for extra PR

**Comment by etrnal70** [comment]
Just tried this PR today. Is the `offset` setting broken ? Because I don't see any change when set it with any value. I'm using wayland on sway

File: docs/dunst.5.pod [hunk 1]
```diff
@@ -64,47 +64,59 @@ keyboard focus.
...
+        top-center
+        top-right
+        bottom-left
+        bottom-center
+        bottom-right
+        left-center
+        center
+        right-center
+
+=item B<offset> format: (horizontal, vertical)
```
Comment (etrnal70) [inline docs/dunst.5.pod:111]: Looks like the format is wrong, should be like `10x300`

**Comment by etrnal70** [comment]
image

Shouldn't hidden notification only show hidden count ? (No icon and progress bar)

File: docs/dunst.5.pod (hunk 1 above)
Comment (fwSmit) [reply docs/dunst.5.pod:111]: Thank you! I've updated it. I think that immediately solves your other problem as well.

**Comment by fwSmit** [comment]
> image
> 
> Shouldn't hidden notification only show hidden count ? (No icon and progress bar)

I'll consider this a separate issue. While investigating this, I also noticed that the timeout of the hidden notifications is also ignored.

**Comment by fwSmit** [comment]
Thanks for testing @etrnal70. I'll go ahead and merge this soon.

File: docs/dunst.5.pod [hunk 2]
```diff
@@ -64,47 +64,59 @@ keyboard focus.
...
 
-It's important to note that the positive and negative sign B<DOES> affect the
-position even if the offset is 0. For example, a horizontal offset of +0 puts
-the notification on the left border of the screen while a horizontal offset of
--0 at the right border. The same goes for the vertical offset.
-
-=back
+Examples:
+        origin = top-right
+        offset = (10, 300) # a margin of 10 pixels from the right and 300 pixels from the top
```
Comment (etrnal70) [inline docs/dunst.5.pod:119]: I forgot this, example should be `10x300`

Comment (fwSmit) [reply docs/dunst.5.pod]: Thank you, it's fixed now

**Pull request merged**